Fix for gmail empty SEARCH responses (#425838 and #420721)

This one comes from the Debian bug report #413336:
http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=413336

diff -r -u isync-1.0.3.orig/src/drv_imap.c isync-1.0.3/src/drv_imap.c
--- isync-1.0.3.orig/src/drv_imap.c	2006-11-01 10:15:09.000000000 +0300
+++ isync-1.0.3/src/drv_imap.c	2007-03-04 06:39:07.000000000 +0300
@@ -865,7 +865,9 @@
 	int uid;
 
 	arg = next_arg( &cmd );
-	if (!arg || !(uid = atoi( arg ))) {
+	if (!arg)
+		return;
+	if (!(uid = atoi( arg ))) {
 		fprintf( stderr, "IMAP error: malformed SEARCH response\n" );
 		return;
 	}
